src/cbc_sdk/workload/vulnerability_assessment.py

```diff
@@ -0,0 +1,902 @@
+
+        Returns:
+            dictinary: monitored_assets and severity summary
+        """
+        query_params = {}
+        if vcenter_id:
+            url = "/vcenters/{}/vulnerabilities/summary".format(vcenter_id)
+        else:
+            url = "/vulnerabilities/summary"
+
+        if severity and severity not in VulnerabilitySummary.VALID_SEVERITY:
+            raise ApiError("Invalid severity")
+        elif severity:
+            query_params["severity"] = severity
+
+        req_url = OrganizationalVulnerability.urlobject.format(
+            cb.credentials.org_key) + url
+        return cb.get_object(req_url, query_params)
+
+
+class Vulnerability(NewBaseModel):
+    """Vulnerability Model"""
+    urlobject = '/vulnerability/assessment/api/v1'
+    urlobject_single = '/vulnerability/assessment/api/v1/vulnerabilities/{}'
+    swagger_meta_file = "workload/models/vulnerability.yaml"
+    url_additional = ""
+    primary_key = "cve_id"
+
+    def __init__(self, cb, model_unique_id, initial_data=None):
+        """
+        Initialize the Vulnerability object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            model_unique_id (str): ID of the alert represented.
+            initial_data (dict): Initial data used to populate the alert.
+        """
+        super(Vulnerability, self).__init__(cb, model_unique_id, initial_data)
+        if model_unique_id is not None and initial_data is None:
+            self._refresh()
+        self._full_init = True
+        self._os_product_id = None
+
+    def _refresh(self):
+        """
+        Rereads the object data from the server.
+
+        Returns:
+            bool: True if refresh was successful, False if not.
+        """
+        resp = self._cb.get_object(self._build_api_request_uri())
+        resp['cve_id'] = self._model_unique_id
+        self._info = resp
+        self._last_refresh_time = time.time()
+        return True
+
+    def _build_api_request_uri(self, http_method="GET"):
+        """
+        Build the unique URL used to make requests for this object.
+
+        Args:
+            http_method (str): Not used; retained for compatibility.
+
+        Returns:
+            str: The URL used to make requests for this object.
+        """
+        return self.urlobject_single.format(self._model_unique_id)
+
+    def affected_assets(self, os_product_id=None):
+        """Returns a list of Vulnerability associated with device.
+
+        Returns:
+            Vulnerability ([Vulnerability]): List of Vulnerability in this device.
+            The os_product_id is required parameter in the request
+        """
+        if os_product_id:
+            self._os_product_id = os_product_id
+            return AffectedAssetQuery(self, self._cb)
+        else:
+            raise ApiError("The os product id is required parameter")
+
+
+class DeviceVulnerability(NewBaseModel):
```

Comment:
This class has no query implementation and returns the API results directly from get request. We often create objects with the results of the response ie
```
for x in results:
  yield Resource(cb, initial_data=x)
```